src/actionlib/action_client.py

```diff
@@ -336,12 +338,15 @@ def update_status(self, status_array):
 
             status = _find_status_by_goal_id(status_array, self.action_goal.goal_id.id)
 
-            # You mean you haven't heard of me?
+            # You mean you haven't heard of me or this is an old goal waiting for status?
             if not status:
                 if self.state not in [CommState.WAITING_FOR_GOAL_ACK,
                                       CommState.WAITING_FOR_RESULT,
                                       CommState.DONE]:
                     self._mark_as_lost()
+                elif self._is_goal_old(status_array.header.stamp) and (
+                     self.state == CommState.WAITING_FOR_GOAL_ACK):
```

Comment:
What about WAIT_FOR_RESULT, could have the same problem if we miss the result msg :(